### PR TITLE
fix data race in tests: ovn mock: deep-copy return values

### PR DESCRIPTION
**Pull request by squeed**
We were returning a pointer in to the mock cache, so tests that asserton values of the cache concurrently trigger races. The test was added in pr #2227

The fix is to deep-copy returned values.

cc @astoycos, who noticed the race.

### CODE REVIEW

File: go-controller/pkg/testing/mock_lrsr.go [hunk 1]
```diff
@@ -3,7 +3,9 @@ package testing
 import (
 	"encoding/base64"
 	"fmt"
+
```
Comment (astoycos) [inline go-controller/pkg/testing/mock_lrsr.go:6]: nit: extra line

**Comment by astoycos** [comment]
Is bringing in `github.com/mitchellh/copystructure` good practice? 

Also I know this works, but shouldn't we be mocking the new sync.Map with a sync map as well (Rather than a normal map with proper locking) ?

otherwise...

/lgtm

**Comment by squeed** [comment]
@astoycos hmm, can you post a traceback from the race detector?

**Comment by trozet** [comment]
> Is bringing in `github.com/mitchellh/copystructure` good practice?
> 
I've never used this copier, but i think it is fine for test code.

> Also I know this works, but shouldn't we be mocking the new sync.Map with a sync map as well (Rather than a normal map with proper locking) ?

It's OK. We aren't trying to test cache functionality here, we are more worried about the consumers of it. So imo sync map vs map with mutex doesn't really matter under the covers.

> 
> otherwise...
> 
> /lgtm

**Review by trozet — APPROVED** [review]

**Pull request merged**